osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/IdentificationNumberFactoryTest.java

```diff
@@ -0,0 +1,116 @@
+
+        assertEquals(last8Digits, identificationNumber.getLast8Digits());
+    }
+
+    @Test
+    public void testFromLast8DigitsDsmr4() {
+
+        String last8Digits = "12049260";
+        Long identification = 302289504L;
+
+        IdentificationNumber identificationNumber = IdentificationNumberFactory.create(
+                createDeviceContext(Protocol.DSMR_4_2_2)).fromLast8Digits(last8Digits);
+
+        assertEquals(identification, identificationNumber.getIdentificationNumber());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromInvalidLast8DigitsDsmr4() {
+
+        IdentificationNumberFactory.create(createDeviceContext(Protocol.DSMR_4_2_2)).fromLast8Digits("123A5678");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromInvalidIdentificationDsmr4() {
+
+        IdentificationNumberFactory.create(createDeviceContext(Protocol.DSMR_4_2_2)).fromIdentification(123456789L);
+    }
+
+    @Test
+    public void testFromIdentificationSmr5() {
+
+        String last8Digits = "12345BEF";
+        Long identification = 305421295L;
+
+        IdentificationNumber identificationNumber = IdentificationNumberFactory.create(
+                createDeviceContext(Protocol.SMR_5_0)).fromIdentification(identification);
+
+        assertEquals(last8Digits, identificationNumber.getLast8Digits());
+    }
+
+    @Test
+    public void testFromLast8DigitsSmr5() {
+
+        String last8Digits = "12345BEF";
+        Long identification = 305421295L;
+
+        IdentificationNumber identificationNumber = IdentificationNumberFactory.create(
+                createDeviceContext(Protocol.SMR_5_0)).fromLast8Digits(last8Digits);
+
+        assertEquals(identification, identificationNumber.getIdentificationNumber());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromInvalidLast8DigitsSmr5() {
+
+        IdentificationNumberFactory.create(createDeviceContext(Protocol.SMR_5_0)).fromLast8Digits("1234S678");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testFromInvalidIdentificationSmr5() {
+
+        IdentificationNumberFactory.create(createDeviceContext(Protocol.SMR_5_0)).fromIdentification(123456789100L);
+    }
+
+    @Test
+    public void testFromNullIdentificationSmr51() {
+
+        IdentificationNumber identificationNumber = IdentificationNumberFactory.create(
+                createDeviceContext(Protocol.SMR_5_1)).fromIdentification(null);
+
+        assertNull(identificationNumber.getLast8Digits());
+    }
+
+    @Test
+    public void testFromNullLast8DigitsSmr51() {
+
+        IdentificationNumber identificationNumber = IdentificationNumberFactory.create(
+                createDeviceContext(Protocol.SMR_5_1)).fromLast8Digits(null);
+
+        assertNull(identificationNumber.getIdentificationNumber());
+    }
+
+    private CommandExecutorDeviceContext createDeviceContext(Protocol protocol) {
```

Comment:
You should not create a real CommandExecutorDeviceContext in this test, but use a mock instead